core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4476,6 +4476,88 @@ public String getDefaultScheme() {
     }
   }
 
+  @Test
+  public void testApplyDefaultServiceConfigIfInitialNameResolutionFails() throws Exception {
+    LoadBalancerRegistry.getDefaultRegistry().register(mockLoadBalancerProvider);
+    timer.forwardNanos(1234);
+    channelBuilder.maxTraceEvents(10);
+    try {
+      Status resolutionError = Status.UNAVAILABLE
+              .withDescription("Initial Name Resolution error, using default service config");
+      FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory
+              .Builder(expectedUri).setServers(Collections.singletonList(new
+              EquivalentAddressGroup(socketAddress))).setError(resolutionError).build();
+
+      channelBuilder.nameResolverFactory(nameResolverFactory);
+
+      Map<String, Object> defaultServiceConfig =
+              parseConfig("{\"methodConfig\":[{"
+                      + "\"name\":[{\"service\":\"SimpleService1\"}],"
+                      + "\"waitForReady\":true}]}");
+
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      createChannel(true);
+
+      int prevSize = getStats(channel).channelTrace.events.size();
+
+      assertThat(getStats(channel).channelTrace.events.get(prevSize - 1))
+              .isEqualTo(new ChannelTrace.Event.Builder()
+              .setDescription("Initial Name Resolution error, using default service config")
+              .setSeverity(ChannelTrace.Event.Severity.CT_ERROR)
+              .setTimestampNanos(timer.getTicker().read())
+              .build());
+
+      nameResolverFactory.resolvers.get(0).listener.onError(resolutionError);
+
+      prevSize = getStats(channel).channelTrace.events.size();
```

Comment:
The reassignment to prevSize makes the below assertThat statement redundant. Remove this reassignment here.

Add a comment that we are asserting there are no new trace events because the lastResolutionState is already ERROR.

Reply:
Fixed